blueocean-dashboard/src/main/js/components/PipelinePage.jsx

```diff
@@ -44,17 +52,18 @@ export default class PipelinePage extends Component {
                         <TabLink to="/pr">Pull Requests</TabLink>
                     </PageTabs>
                 </PageHeader>
-                {React.cloneElement(this.props.children, { pipeline })}
+                {this.props.children}
             </Page>
         );
     }
 }
 
 PipelinePage.propTypes = {
-    children: object,
+    children: PropTypes.any,
```

Comment:
as you said "code style difference of opinion". As stated it is the context, but I agree to move it just before the usage makes a lot of sense. 

Most of you points are covered by eslint rules and I am against using React.PropTypes.<type> there is no benefit in it You will repeat React.PropTypes meaningless times. 

Anyway ....